test/testcmdlineparser.cpp

```diff
@@ -104,6 +104,19 @@ class TestCmdlineParser : public TestFixture {
         // TODO: verify that the redirect output is empty
     }
 
+    // add specialization so the enums can be compared without a cast
+    template<typename T, typename U>
+    bool assertEquals(const char* const filename, const unsigned int linenr, const T& expected, const U& actual) const
+    {
+        return TestFixture::assertEquals(filename, linenr, expected, actual);
+    }
+
+    template<>
+    bool assertEquals<CmdLineParser::Result, CmdLineParser::Result>(const char* const filename, const unsigned int linenr, const CmdLineParser::Result& expected, const CmdLineParser::Result& actual) const
```

Comment:
I added the specialization so I didn't have to introduce hundreds of `static_cast` into the asserts.

Reply:
If we find a way to simply define this specialization we can clean up other tests.